### PR TITLE
adds e2e CI test for hypertrace-service

### PR DESCRIPTION
**Pull request by JBAhire**
This PR,
- will add quick e2e test for hypertrace-service which will 
   - Build image on each new PR and push to main branch
   - test if we are able to get full stack up and running
   - generate and retrieve trace.

### CODE REVIEW

**Comment by jcchavezs** [comment]
I like this as a first approach, great work! In the future I'd like to see this depending on less components (e.g. writing directly to the topics) as that would ease the digestion of the data flowing through this specific component. Also, maybe it is a good idea to start by generating data directly zipkin endpoint instead of using the apps? cc @adriancole

File: .github/workflows/docker-tests.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+name: e2e test
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  # workflow_dispatch will let us manually trigger the workflow from GitHub actions dashboard.
```
Comment (adriancole) [inline .github/workflows/docker-tests.yml:7]:
```suggestion
  # workflow_dispatch will let us manually trigger the workflow from GitHub actions dashboard.
  # See https://docs.github.com/en/free-pro-team@latest/actions/managing-workflow-runs/manually-running-a-workflow
```

File: .github/workflows/docker-tests.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        uses: actions/cache@v1
+        with:
+          path: ~/.gradle
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/build.gradle.kts') }}
+          restore-keys: ${{ runner.os }}-gradle
+      - name: Invoke Docker build with Gradle
+        uses: eskatos/gradle-command-action@v1
+        with:
+          arguments: dockerBuildImages
+      - name: verify docker image creation
```
Comment (adriancole) [inline .github/workflows/docker-tests.yml:50]: remove this as no longer debugging the job

File: .github/workflows/hypertrace-service/scripts/tests.sh [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+  RETRY_COUNT=$((RETRY_COUNT+1)) 
+  sleep 3
+done
+
+echo ""
+echo "Found trace \"$TRACE_ID\"."
+
+echo "Checking for broken links"
+
+curl http://${TRACES_SERVER_HOST}:2020/ -s -f -o /dev/null   || { echo "home link is not available" ; exit 1; }
```
Comment (adriancole) [inline .github/workflows/hypertrace-service/scripts/tests.sh:98]: when this merges, remove these checks from hypertrace/hypertrace DRY

**Review by adriancole — APPROVED** [review]

**Review by buchi-busireddy — APPROVED** [review]

**Pull request merged**